### PR TITLE
Rename plug-ins to apps and move them to home

### PR DESCRIPTION
**Pull request by mdmohsin7**
Related to #1067 
This PR also refactors the plugins/apps page to use selectors instead of a consumer for the whole page to reduce widget rebuilds

https://github.com/user-attachments/assets/e6f3526e-ebf8-420e-9b9a-f51e2600b742

## Summary by CodeRabbit

- **New Features**
	- Introduced a new "Apps" tab in the navigation structure, enhancing user access to applications.
	- Updated terminology throughout the app, replacing "Plugins" with "Apps" for consistency.
	- Added a dedicated AppsPage to display applications categorized into "Memories" and "Chat".
	- Implemented a new AppListItem widget for displaying individual applications.

- **Bug Fixes**
	- Improved UI clarity and functionality in various components, including dropdowns and memory detail widgets.

- **Refactor**
	- Streamlined interface for `AppListItem` and enhanced state management in `AppsPage`.
	- Transitioned from PluginProvider to AppProvider across the application for better management of app data.
	- Updated the DeleteAccount widget to reflect changes in terminology.
	- Renamed and refactored classes and methods to align with the new app-based architecture.
	- Enhanced JSON handling in the Structured class for improved data integrity.
	- Adjusted the ChatPage and related components to work with the new app data model.

- **Chores**
	- Removed the "Plugins" section from the SettingsPage for a cleaner user experience.
	- Removed unused import statements for better code cleanliness.

### CODE REVIEW

File: app/lib/pages/plugins/page.dart
Comment (adamcohenhillel) [inline app/lib/pages/plugins/page.dart:1]: I guess we should name the folder to "apps" too? and then change all the improts

**Comment by kodjima33** [comment]
UI looks great, if code is good, please merge ASAP

File: app/lib/pages/plugins/page.dart
Comment (mdmohsin7) [reply app/lib/pages/plugins/page.dart:1]: Can do, didn't do it now itself because we still have the plugins terminology in the backend as well

Comment (kodjima33) [reply app/lib/pages/plugins/page.dart:1]: yeah everything should be apps. plugins word should be removed

**Comment by mdmohsin7** [comment]
Renamed it everywhere except for:
- Mixpanel analytics

I need some clarity on whether to rename for mixpanel as well (will result in new events)

**Comment by beastoin** [comment]
> Renamed it everywhere except for:
> 
>     * Mixpanel analytics
> 
> 
> I need some clarity on whether to rename for mixpanel as well (will result in new events)

no worry, rename it as well, please sir; Mixpanel should support the mixing events? if not just moving on and tracking the new event.

**Comment by beastoin** [comment]
lgtm 🚀 after relesing please inform team whom use Mixpannel
about changes on event. we have to keep the term as simple as posible, so next the backend/db should use the new term: app, as well.

**Pull request merged**